dotCMS/src/main/java/com/dotcms/util/ContentTypeUtil.java

```diff
@@ -146,40 +151,89 @@ public String getActionUrl( HttpServletRequest request, final ContentType conten
-							contentTypeInode, user.getUserId(), e.getMessage()),
-					e);
+            Logger.error(this,
+                    String.format(
+                            "An error occurred when retrieving the action URL of Content Type [%s] for user [%s]: %s",
+                            contentTypeInode, user.getUserId(), e.getMessage()),
+                    e);
         }
         return actionUrl;
     }
 
-	/**
+    /**
+     * Generates the referrer URL that will indicate the system what location the user will be
+     * redirected to after performing an operation in the back-end. For example, this can be used by
+     * the "+" sign component that adds different types of content to the system, as it indicates
+     * where to return after adding new content.
+     * 
+     * @param request - The {@link HttpServletRequest} object.
+     * @param layout - The layout where the user will be redirected after performing his task.
+     * @param contentTypeInode - The Inode of the content type used by the new content.
+     * @param user - The user performing this action.
+     * @return The referrer URL.
+     * @throws WindowStateException If the portlet does not support the
+     *         {@link WindowState.MAXIMIZED} state.
+     */
+    private String generateReferrerUrl(final HttpServletRequest request, final Layout layout, final String contentTypeInode,
+                    final User user) throws WindowStateException {
+        final PortletURL portletURL = new PortletURLImpl(request, PortletID.CONTENT.toString(), layout.getId(), true);
+        portletURL.setWindowState(WindowState.MAXIMIZED);
+        portletURL.setParameters(map(
+                        "struts_action", new String[] {"/ext/contentlet/view_contentlets"}, 
+                        "cmd", new String[] {"new"}, 
+                        "inode", new String[] {""}, 
+                        "structure_id", new String[] {contentTypeInode}, 
+                        "lang", new String[] {this.getLanguageId(user.getLanguageId()).toString()}));
+        return portletURL.toString();
+    }
+
+    /**
+     * Traverses the list of {@link Layout} objects in order to find the one that contains the
+     * "Content" portlet. If not present, then the first layout in the list is returned by default.
+     * Layouts are basically the menu option that groups one or more links that lead to portlets in
+     * dotCMS.
+     * 
+     * @param layouts - The list of layouts.
+     * @return The specific layout that contains the "Content" portlet.
+     */
+    private Layout getContentPortletLayout(final List<Layout> layouts) {
+        for (Layout layout : layouts) {
+            final List<String> portletIds = layout.getPortletIds();
+            if (UtilMethods.isSet(portletIds)) {
+                final Optional<String> contentPortletId =
+                                portletIds.stream().filter(id -> id.equals(PortletID.CONTENT.toString())).findFirst();
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Specify a type for: 'id' [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2211)